app/src/main/java/org/oppia/android/app/devoptions/platformparameters/PlatformParametersFragmentPresenter.kt

```diff
@@ -123,62 +126,63 @@ class PlatformParametersFragmentPresenter @Inject constructor(
         editText.inputType = InputType.TYPE_CLASS_NUMBER
         val displayValue = when (val storedValue = paramState?.integer) {
           -1 -> {
-            model.errorMessage.set(invalidInputErrorText)
+            model.inputErrorMsg.set(invalidInputErrorText)
             ""
           }
-
-          null -> model.currentValue.integer.toString()
-          else -> storedValue.toString()
+          null -> {
+            model.inputErrorMsg.set("")
+            model.currentValue.integer.toString()
+          }
+          else -> {
+            model.inputErrorMsg.set("")
+            storedValue.toString()
+          }
         }
         model.inputValue.set(displayValue)
       }
 
       model.currentValue.hasString() -> {
         editText.inputType = InputType.TYPE_CLASS_TEXT
-        if (paramState != null) {
-          model.inputValue.set(paramState.string)
-        }
-      }
 
-      else -> {
-        editText.inputType = InputType.TYPE_CLASS_TEXT
+        model.inputValue.set(paramState?.string ?: model.currentValue.string)
+        model.inputValue.set("")
```

Comment:
inputErrorMsg?

```suggestion
        model.inputErrorMsg.set("")
```

model.inputValue.set("") should be set for fallback else case, which seems to be removed now.

Reply:
updated